homeassistant/components/image_processing/facebox.py

```diff
@@ -94,28 +154,53 @@ def __init__(self, ip, port, camera_entity, name=None):
 
     def process_image(self, image):
         """Process an image."""
-        response = {}
-        try:
-            response = requests.post(
-                self._url,
-                json={"base64": encode_image(image)},
-                timeout=TIMEOUT
-                ).json()
-        except requests.exceptions.ConnectionError:
-            _LOGGER.error("ConnectionError: Is %s running?", CLASSIFIER)
-            response['success'] = False
-
-        if response['success']:
-            total_faces = response['facesCount']
-            faces = parse_faces(response['faces'])
-            self._matched = get_matched_faces(faces)
-            self.process_faces(faces, total_faces)
+        response = post_image(self._url_check, image)
+        if response is not None:
+            response_json = response.json()
+            if response_json['success']:
+                total_faces = response_json['facesCount']
+                faces = parse_faces(response_json['faces'])
+                self._matched = get_matched_faces(faces)
+                self.process_faces(faces, total_faces)
 
         else:
             self.total_faces = None
             self.faces = []
             self._matched = {}
 
+    def teach(self, name, file_path):
+        """Teach classifier a face name."""
+        if (self.hass.config.is_allowed_path(file_path) and
```

Comment:
Make this a guard clause by inverting the check and returning if true. Then you can outdent the following code. :tada:

Reply:
ok